File: testapp/src/main/java/com/appliedrec/mrtd_reader_app/FaceComparisonActivity.java
package com.appliedrec.mrtd_reader_app;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.os.Bundle;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.drawable.RoundedBitmapDrawable;
import androidx.core.graphics.drawable.RoundedBitmapDrawableFactory;

import com.appliedrec.mrtd_reader_sample.R;
import com.appliedrec.mrtd_reader_sample.databinding.ActivityFaceComparisonBinding;

import org.apache.commons.math3.distribution.NormalDistribution;

public class FaceComparisonActivity extends AppCompatActivity {

    public static final String EXTRA_SCORE = "score";
    public static final String EXTRA_IMAGE1 = "image1";
    public static final String EXTRA_IMAGE2 = "image2";
    float threshold = 3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityFaceComparisonBinding viewBinding = ActivityFaceComparisonBinding.inflate(getLayoutInflater());
        setContentView(viewBinding.getRoot());
        float score = getIntent().getFloatExtra(EXTRA_SCORE, 0f);
        if (score >= threshold) {
            NormalDistribution normalDistribution = new NormalDistribution();
            double probability = normalDistribution.cumulativeProbability(score)*100;
            viewBinding.contextLabel.setText(getString(R.string.pass_details, score, probability, threshold));
            viewBinding.resultStatusTextView.setText(R.string.pass);
            int green = Color.rgb(54, 175, 0);
            viewBinding.resultStatusTextView.setTextColor(green);
        } else {
            viewBinding.contextLabel.setText(getString(R.string.warning_details, score, threshold));
            viewBinding.resultStatusTextView.setText(R.string.warning);
            viewBinding.resultStatusTextView.setTextColor(Color.RED);
        }
        ImageView[] imageViews = new ImageView[]{
                viewBinding.face1ImageView,
                viewBinding.face2ImageView
        };
        byte[] image1 = getIntent().getByteArrayExtra(EXTRA_IMAGE1);
        byte[] image2 = getIntent().getByteArrayExtra(EXTRA_IMAGE2);
        if (image1 != null && image2 != null) {
            byte[][] images = new byte[2][0];
            images[0] = image1;
            images[1] = image2;
            for (int i=0; i<images.length; i++) {
                Bitmap bitmap = BitmapFactory.decodeByteArray(images[i], 0, images[i].length);
                RoundedBitmapDrawable drawable = RoundedBitmapDrawableFactory.create(getResources(), bitmap);
                drawable.setCornerRadius((float)drawable.getIntrinsicWidth()/8f);
                imageViews[i].setImageDrawable(drawable);
            }
        }
    }
}

File: testapp/src/main/java/com/appliedrec/mrtd_reader_app/FaceComparisonActivity.kt
package com.appliedrec.mrtd_reader_app

import android.graphics.BitmapFactory
import android.graphics.Color
import android.os.Bundle
import androidx.annotation.StringRes
import androidx.appcompat.app.AppCompatActivity
import androidx.core.graphics.drawable.RoundedBitmapDrawableFactory
import com.appliedrec.mrtd_reader_app.databinding.ActivityFaceComparisonBinding
import org.apache.commons.math3.distribution.NormalDistribution

class FaceComparisonActivity : AppCompatActivity() {

    companion object {
        const val EXTRA_SCORE = "score"
        const val EXTRA_IMAGE1 = "image1"
        const val EXTRA_IMAGE2 = "image2"
        const val EXTRA_THRESHOLD = "threshold"
        const val EXTRA_GLASSES_DETECTED = "glasses_detected"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val viewBinding = ActivityFaceComparisonBinding.inflate(
            layoutInflater
        )
        setContentView(viewBinding.root)
        val score = intent.getFloatExtra(EXTRA_SCORE, 0f)
        val threshold = intent.getFloatExtra(EXTRA_THRESHOLD, 0.6f)
        val glassesDetected = intent.getBooleanExtra(EXTRA_GLASSES_DETECTED, false)
        val warningThreshold = threshold - 0.1f
        if (score >= threshold) {
            viewBinding.contextLabel.text = getString(R.string.pass_details, score, threshold)
            viewBinding.resultStatusTextView.setText(R.string.pass)
            val green = Color.rgb(54, 175, 0)
            viewBinding.resultStatusTextView.setTextColor(green)
        } else if (score >= warningThreshold) {
            @StringRes val resId: Int = if (glassesDetected) R.string.warning_details_glasses else R.string.warning_details
            viewBinding.contextLabel.text = getString(resId, score, threshold)
            viewBinding.resultStatusTextView.setText(R.string.warning)
            viewBinding.resultStatusTextView.setTextColor(Color.rgb(244, 191, 79))
        } else {
            @StringRes val resId: Int = if (glassesDetected) R.string.fail_details_glasses else R.string.fail_details
            viewBinding.contextLabel.text = getString(resId, score, threshold)
            viewBinding.resultStatusTextView.setText(R.string.fail)
            viewBinding.resultStatusTextView.setTextColor(Color.RED)
        }
        val imageViews = arrayOf(
            viewBinding.face1ImageView,
            viewBinding.face2ImageView
        )
        val image1 = intent.getByteArrayExtra(EXTRA_IMAGE1)
        val image2 = intent.getByteArrayExtra(EXTRA_IMAGE2)
        if (image1 != null && image2 != null) {
            val images = Array(2) { ByteArray(0) }
            images[0] = image1
            images[1] = image2
            for (i in images.indices) {
                val bitmap = BitmapFactory.decodeByteArray(images[i], 0, images[i].size)
                val drawable = RoundedBitmapDrawableFactory.create(resources, bitmap)
                drawable.cornerRadius = drawable.intrinsicWidth.toFloat() / 8f
                imageViews[i].setImageDrawable(drawable)
            }
        }
    }
}